### PR TITLE
WIP Extract userdata to s3 bucket

### PR DESCRIPTION
**Pull request by hectorhuertas**

### CODE REVIEW

**Comment by ffilippopoulos** [comment]
lgtm. My only objection/question is: shall we encrypt the stuff that we store in the bucket? since its a public bucket

**Comment by hectorhuertas** [comment]
Initially we could extract only files that have no sensitive data. If we find that's not enough, we could also close the bucket to be only accessible from our instances

**Comment by george-angel** [comment]
@ffilippopoulos is there an option to encrypt it? Remember we aren't fetching this data ourselves, ignition does.

File: variables.tf [hunk 1]
```diff
@@ -174,3 +174,6 @@ variable "cfssl_data_volumeid" {}
 variable "etcd_data_volumeids" {
   type = "list"
 }
+
+variable "account" {}
```
Comment (george-angel) [inline variables.tf]: What is account in this instance?

Comment (hectorhuertas) [reply variables.tf]: dev / prod. It is needed because bucket names are global

Comment (ffilippopoulos) [reply variables.tf]: so it could also be environment? `${var.environment}`

**Comment by hectorhuertas** [comment]
Closing in favor of https://github.com/utilitywarehouse/terraform/pull/831 to keep this module provider agnostic

**Pull request closed without merging**